api/integration/registry/license_providers.py

```diff
@@ -6,13 +6,10 @@
 from core.integration.registry import IntegrationRegistry
 
 if TYPE_CHECKING:
-    from api.circulation import BaseCirculationAPI  # noqa: autoflake
-    from core.integration.settings import BaseSettings  # noqa: autoflake
+    from api.circulation import CirculationApiType  # noqa: autoflake
 
 
-class LicenseProvidersRegistry(
-    IntegrationRegistry["BaseCirculationAPI[BaseSettings, BaseSettings]"]
-):
+class LicenseProvidersRegistry(IntegrationRegistry["CirculationApiType"]):
```

Comment:
Unfortunately future annotations doesn't cover the parameters for generic classes, so we still need to use a string annotation here.